specutils/io/default_loaders/manga.py

```diff
@@ -0,0 +1,177 @@
+
+import astropy.units as u
+from astropy.io import fits
+from astropy.nddata import InverseVariance
+
+import _io
+import contextlib
+from ...spectra import Spectrum1D
+from ..registers import data_loader
+
+__all__ = ["identify_manga_cube", "identify_manga_rss", "manga_cube_loader", "manga_rss_loader"]
+
+
+@contextlib.contextmanager
+def _read_fileobj(*args, **kwargs):
+    """ Context manager for reading a filename or file object
+
+    Returns:
+        an Astropy HDUList
+    """
+    # access the fileobj or filename arg
+    # do this so identify functions are useable outside of Spectrum1d.read context
+    try:
+        fileobj = args[2]
+    except IndexError:
+        fileobj = args[0]
+
+    if isinstance(fileobj, fits.hdu.hdulist.HDUList):
+        hdulist = fileobj
+    elif isinstance(fileobj, _io.BufferedReader):
+        hdulist = fits.open(fileobj)
+    else:
+        hdulist = fits.open(fileobj, **kwargs)
+
+    yield hdulist
+
+    if not isinstance(fileobj, (fits.hdu.hdulist.HDUList, _io.BufferedReader)):
+        hdulist.close()
+
+
+def _identify_sdss_fits(*args, **kwargs):
+    """
+    Check whether the given file is a SDSS data product.
+    """
+
+    try:
+        with _read_fileobj(*args, **kwargs) as hdulist:
+            return hdulist[0].header["TELESCOP"] == "SDSS 2.5-M"
+    except Exception:
+        return False
+
+
+def identify_manga_cube(origin, *args, **kwargs):
+    """
+    Check whether the given file is a MaNGA CUBE.
+    """
+
+    is_sdss = _identify_sdss_fits(*args, **kwargs)
+    with _read_fileobj(*args, **kwargs) as hdulist:
+        return (is_sdss and "FLUX" in hdulist and hdulist[1].header['INSTRUME'] == 'MaNGA'
+                and hdulist[1].header["NAXIS"] == 3)
+
+
+def identify_manga_rss(origin, *args, **kwargs):
+    """
+    Check whether the given file is a MaNGA RSS.
+    """
+    is_sdss = _identify_sdss_fits(*args, **kwargs)
+    with _read_fileobj(*args, **kwargs) as hdulist:
+        return (is_sdss and "FLUX" in hdulist and hdulist[1].header['INSTRUME'] == 'MaNGA'
```

Comment:
Wondering if it's really worth delegating the `is_sdss` check to its own function, as the direct `hdulist[0].header["TELESCOP"] == "SDSS 2.5-M"` test would fit almost within the same space.
Since it is obviously working, it should be OK, but we might want to avoid extra file I/O operations from a performance point of view.

Reply:
I agree.  I've removed the redundant function and moved the the SDSS telescope check into the `cube` and `rss` identifiers